gfx/video_driver.c

```diff
@@ -2525,8 +2528,34 @@ void video_driver_build_info(video_frame_info_t *video_info)
    video_info->runloop_is_slowmotion       = runloop_st->flags & RUNLOOP_FLAG_SLOWMOTION;
    video_info->fastforward_frameskip       = settings->bools.fastforward_frameskip;
 
-   video_info->input_driver_nonblock_state = input_st
-         ? (input_st->flags & INP_FLAG_NONBLOCKING) : false;
+#ifdef _WIN32
+#ifdef HAVE_VULKAN
+   /* Vulkan in Windows does mailbox emulation
+    * in fullscreen with vsync, effectively
+    * discarding frames that can't be shown,
+    * therefore do not do it twice. */
+   if (     string_is_equal(video_driver_get_ident(), "vulkan")
```

Comment:
What if we instead move this to the Vulkan video frame function in gfx/drivers/vulkan.c?

Reply:
How? Vulkan already isn't doing it when vsync is disabled, and it can't just be disabled always without negative side effects.